test/functional/test_framework/authproxy.py

```diff
@@ -139,17 +139,24 @@ def _request(self, method, path, postdata):
             return self._get_response()
 
     def __call__(self, *args, **argsn):
-        AuthServiceProxy.__id_count += 1
+        # check for a possible wallet endpoint
+        urlAdd = ""
+        method = self._service_name
+        elements = self._service_name.split(".")
+        if len(elements) > 1:
+            urlAdd = "/wallet/" + '.'.join(elements[:-1])
```

Comment:
In commit "[QA] Add support for wallet endpoints in Authproxy"

It seems hacky to me for AuthServiceProxy to be aware of multiwallet and to be munging urls and service name strings this way.

I think it would be better if AuthServiceProxy didn't know anything about bitcoin urls and just let the caller control the request. There are many ways to allow this, but a simple one might be to define:

```python
def __idiv__(self, relative_uri):
    return AuthServiceProxy("{}/{}".format(self.__service_url, relative_uri), self._service_name, connection=self.__conn)
```

Then you could call methods on individual wallets with:

```python
w1 = self.nodes[0] / "wallet/w1.dat"
w1.getwalletinfo()
w1.getbalance()
```